src/marshmallow/fields.py

```diff
@@ -1281,28 +1281,23 @@ def _serialize(self, value, attr, obj, **kwargs) -> str | float | None:
         format_func = self.SERIALIZATION_FUNCS.get(data_format)
         if format_func:
             return format_func(value)
-        else:
-            return value.strftime(data_format)
+        return value.strftime(data_format)
 
     def _deserialize(self, value, attr, data, **kwargs) -> dt.datetime:
         if not value:  # Falsy values, e.g. '', None, [] are not valid
             raise self.make_error("invalid", input=value, obj_type=self.OBJ_TYPE)
         data_format = self.format or self.DEFAULT_FORMAT
         func = self.DESERIALIZATION_FUNCS.get(data_format)
-        if func:
-            try:
-                return func(value)
-            except (TypeError, AttributeError, ValueError) as error:
-                raise self.make_error(
-                    "invalid", input=value, obj_type=self.OBJ_TYPE
-                ) from error
-        else:
-            try:
-                return self._make_object_from_format(value, data_format)
-            except (TypeError, AttributeError, ValueError) as error:
-                raise self.make_error(
-                    "invalid", input=value, obj_type=self.OBJ_TYPE
-                ) from error
+        if not func:
+            func = functools.partial(
```

Comment:
(non-blocking) is there any performance hit from creating this partial?

Reply:
Good point. Didn't check.

We can just revert and move the try/catch out of the `if func` block. This is a totally acceptable scope.